File: Queue & Stack/CPP/backspace_string_compare.cpp
class Solution {
public:
    bool backspaceCompare(string s, string t) {
        return st(s) == st(t);
    }

    string st(string s) {
        string curs;
        for (char e : s) {
            if (e != '#') curs.push_back(e); 
            else if (!curs.empty()) curs.pop_back();
        }
        return curs;
    }
};

File: Queue & Stack/backspace_string_compare.py
class Solution:
    """
    Task:
        Given two strings s and t, return true if they are equal when both are typed into empty text editors.
        '#' means a backspace character.

        Note that after backspacing an empty text, the text will continue empty.
    """
    def backspaceCompare(self, s: str, t: str) -> bool:
        def stack(s):
            j = ""
            for e in s:
                if e != '#': j += e
                else:
                    if j: j = j[:-1]
            return j
        return stack(s) == stack(t)